src/components/ConnectButton/WalletDropdownButton.tsx

```diff
@@ -0,0 +1,66 @@
+"use client";
+import { useState } from "react";
+import { useRouter } from "next/navigation";
+import { useAccount, useDisconnect } from "wagmi";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger
+} from "@/components/ui/dropdown-menu";
+import { Icons } from "../Icons";
+import NoSSR from "../NoSSR";
+import { Button } from "@/components/ui/button";
+import ConnectButton from ".";
+import { truncateAddress } from "@/lib/utils";
+import useClipboard from "@/hooks/useClipboard";
+
+export default function WalletDropdownButton() {
+  const { address } = useAccount();
+  const { disconnect } = useDisconnect();
+  const { push } = useRouter();
+  const { setClipboard } = useClipboard();
+  const [isCopied, setIsCopied] = useState(false);
+
+  const onCopyAddress = () => {
+    if (!address) return;
```

Comment:
if there is no address, the component should never render right?
- when would this check be helpful?
- can this check be removed?

rules
- less is more
- simple is best
- small is beautiful